controller/helper/process-patient-status.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+/**
+ * @module controller/handler/trial
+ */
+
+/**
+ * A dashboard with an overview of a specific trial.
+ * @param {Array<Object>} rows - aggregated survey information
+ * @returns {Array<Object>} rows - updated rows object with compliance status set for each record
+ */
+function processPatientStatus (rows) {
+    const redThreshold = 2;
+    const yellowThresholdBegin = 0;
+
+    for (const index in rows) {
```

Comment:
since this only needs one data source (rows) calling `Array.map` from the controller.